### PR TITLE
Add onion3 multiaddr support

### PR DESCRIPTION
**Pull request by diegomrsantos**
Add onion3 multiaddr support

### CODE REVIEW

File: libp2p/multiaddress.nim [hunk 1]
```diff
@@ -363,6 +402,10 @@ const
       mcodec: multiCodec("onion"), kind: Fixed, size: 10,
```
Comment (Menduist) [inline libp2p/multiaddress.nim:402]: Can you fix this to 12 while we're at it?

Comment (diegomrsantos) [reply libp2p/multiaddress.nim:402]: Thought about that, but is it fine to do that in this PR?

Comment (Menduist) [reply libp2p/multiaddress.nim:402]: Yes go ahead, better than doing a PR for 1 LoC imo

**Comment by Menduist** [comment]
Looks good!

File: libp2p/multiaddress.nim (hunk 1 above)
Comment (diegomrsantos) [reply libp2p/multiaddress.nim:402]: Maybe tests for the fix should also be added?

Comment (Menduist) [reply libp2p/multiaddress.nim:402]: Yes good point, maybe we should do another PR with:
- The fix
- Roundtrip tests (need to replace `ipfs` by `p2p` in the test inputs)
- Roundtrip tests "per part": `ma[0..^1].tryGet().mapIt($it).join('')` or something like that
The last one would have caught the bug in the first place

**Review by Menduist — DISMISSED** [review]

**Review by Menduist — APPROVED** [review]

**Review by Menduist — APPROVED** [review]

**Pull request merged**